### DISCUSSION TITLE
How are print dimensions calculated in UV Tools (not sure if this should be a bug report)

### DISCUSSION THREAD

**Discussion by cunabula27** [sn4k3/UVtools#684] [Q&A] [answered]
I have this layer which is 280px square and on my printer a pixel is 28.5um. UV Tools recognises both of these and says the layer should come out at 8.12mm square but 280 x .0285 is 7.98 not 8.12. 

Is there something I'm missing that makes this difference make sense? 

Or do you think this is a bug?

![image](https://user-images.githubusercontent.com/24782488/228301227-4df57458-c82f-4004-a288-78520d18c9de.png)

Edit: I've just done the obvious reverse calculation and it appears to be a rounding issue. I think UV Tools is using 0.029 for the pixel calculation.

**Comment by sn4k3** [comment, marked as answer]
Most likely is a rounding, calculation can be found here:

https://github.com/sn4k3/UVtools/blob/df7d434284d5d14030d893400cfa7121a1adb779/UVtools.Core/FileFormats/FileFormat.cs#L1751-L1764
https://github.com/sn4k3/UVtools/blob/df7d434284d5d14030d893400cfa7121a1adb779/UVtools.Core/Layers/Layer.cs#L214-L225

To have better precision I can increase the rounding to 4 on the `PixelSize`

**Comment by cunabula27** [comment]
Thank you for the quick reply. I was working on calibrating my printer and that was making me scratch my head a bit.
